pkg/file/file.go

```diff
@@ -193,3 +196,67 @@ func ExtractFilenameFromURLPath(p string, defaultExtension string) string {
 
 	return b
 }
+
+// ExtractTarGz extracts the contents of the provided tar.gz file.
+// This implementation uses os.* calls without permission checks, as we're
+// running this operation in the context of fleetd running as root on a host
+// (e.g. for installs), so we have different constraints than fleetctl building
+// a package.
+func ExtractTarGz(path string, destDir string) error {
+	tarGzFile, err := secure.OpenFile(path, os.O_RDONLY, 0o755)
```

Comment:
Do we need to check permissions when opening the `tar.gz`? (Given `path`  is created by orbit with òs.MkdirTemp` which creates the directory with `0o700`)

Reply:
Sorry, I thought I swapped this to `os.Open`, which calls OpenFile with the proper args under the hood. Fixing now.